### PR TITLE
[BugFix] The update_config interface still returns success after the dynamic parameter setup fails.

### LINKED ISSUE #50822: The update_config interface still returns success after the dynamic parameter setup fails
### Steps to reproduce the behavior (Required)
max_threads=64
min_threads=0
```
curl -XPOST http://be_host:http_port/api/update_config?transaction_apply_thread_pool_num_min=1000
```

### Expected behavior (Required)
1 update_config return Err
2 min_threads=0
### Real behavior (Required)
1 update_config return OK
2 min_threads=1000

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by zhangheihei**
## What I'm doing:

Fixes #50822

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/common/configbase.h [hunk 1]
```diff
@@ -160,6 +170,9 @@ class Field {
     void* _storage;
     const char* _defval;
     bool _valmutable;
+    char* _last_set_val;
```
Comment (dirtysalt) [inline be/src/common/configbase.h:173]: Does it has to be c-string? why not std::string?  

if it's std::string, code looks more neat.

Comment (zhangheihei) [reply be/src/common/configbase.h]: tks review.
I had updated c style to c++

File: be/src/common/configbase.cpp [hunk 2]
```diff
@@ -189,7 +189,23 @@ bool Field::set_value(std::string value) {
         return false;
     }
     StripWhiteSpace(&value);
-    return parse_value(value);
+    bool flag = parse_value(value);
+    if (flag) {
+        _current_set_val = value;
+    }
+    return flag;
```
Comment (kevincai) [inline be/src/common/configbase.cpp:196]:
```suggestion
    bool success = parse_value(value);
    if (success) {
        _last_set_val.swap(_current_set_val);
        _current_set_val = value;
    }
    return success;
```

File: be/src/common/configbase.cpp [hunk 3]
```diff
@@ -189,7 +189,23 @@ bool Field::set_value(std::string value) {
...
+}
+
+void Field::set_last_update_value() {
+    _last_set_val = _current_set_val;
+}
+
+bool Field::rollback_last_value() {
+    bool flag = parse_value(_last_set_val);
+    if (flag) {
+        _current_set_val = _last_set_val;
```
Comment (kevincai) [inline be/src/common/configbase.cpp:206]: need to reset _last_set_val, `_last_set_val.clear()`.
And when doing the rollback, need to check if _last_set_val is empty or not, so the rollback can't be performed recursively.

File: be/src/common/configbase.cpp [hunk 4]
```diff
@@ -189,7 +189,23 @@ bool Field::set_value(std::string value) {
...
     StripWhiteSpace(&value);
-    return parse_value(value);
+    bool flag = parse_value(value);
+    if (flag) {
+        _current_set_val = value;
+    }
+    return flag;
+}
+
+void Field::set_last_update_value() {
```
Comment (kevincai) [inline be/src/common/configbase.cpp:199]: interface is not necessary.

File: be/src/common/configbase.h [hunk 5]
```diff
@@ -142,6 +142,10 @@ class Field {
 
     bool set_value(std::string value);
 
+    bool rollback_last_value();
```
Comment (kevincai) [inline be/src/common/configbase.h:145]:
```suggestion
    bool rollback();
```

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**